internal/protection/usage.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2025 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package protection contains API types that protect resources from deletion.
+package protection
+
+import (
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+// AnnotationKeyDeletionAttempt is the annotation key used to record whether
+// a deletion attempt was made and blocked by the Usage. The value stored is
+// the propagation policy used with the deletion attempt.
+const AnnotationKeyDeletionAttempt = "usage.crossplane.io/deletion-attempt-with-policy"
+
+// ResourceRef is a reference to a resource.
+type ResourceRef struct {
+	// Name of the referent.
+	Name string
+
+	// Namespace of the referent.
+	// Omit for cluster scoped referents, or referents in the same namespace.
+	Namespace *string
+}
+
+// ResourceSelector is a selector to a resource.
+type ResourceSelector struct {
+	// MatchLabels ensures an object with matching labels is selected.
+	MatchLabels map[string]string
+
+	// MatchControllerRef ensures an object with the same controller reference
+	// as the selecting object is selected.
+	MatchControllerRef *bool
+
+	// Namespace of the referent.
+	// Omit for cluster scoped referents, or to match referents across all
+	// namespaces.
```

Comment:
I need to update this to match the comment above on `ResourceRef.Namespace`.

I originally thought it'd match across all namespaces when omitted, but decided it was better to match within the Usage's namespace (if the Usage is namespaced).